onadata/apps/api/viewsets/xform_viewset.py

```diff
@@ -423,24 +421,22 @@ def enketo(self, request, **kwargs):
             request_vars = request.GET
             defaults = generate_enketo_form_defaults(
                 self.object, **request_vars)
-            url = enketo_url(
+            enketo_urls = get_enketo_urls(
                 form_url, self.object.id_string, **defaults)
-            preview_url = get_enketo_preview_url(request,
-                                                 self.object.user.username,
-                                                 self.object.id_string,
-                                                 xform_pk=self.object.pk)
+            offline_url = enketo_urls.get('offline_url')
+            preview_url = enketo_urls.get('preview_url')
+            single_submit_url = enketo_urls.get('single_url')
         except EnketoError as e:
             data = {'message': _(u"Enketo error: %s" % e)}
         else:
             if survey_type == 'single':
```

Comment:
Are there any other `survey_type` query params we support?

Reply:
_I think it's a bit much to have a query param that just has one supported value 😅  But that might just be me..._

Reply:
It was intentional when defining this endpoint initially.
I did not see anything wrong with leaving that there, though we could remove that.
However there is already published documentation on this was of querying the endpoint that might also need to be scraped.
Im unsure of the need to really remove this here.

Reply:
No need to remove it... Just wondering if we should support the `preview` and `offline` URL

Reply:
That i something I believe we could potentially consider

Reply:
Nice, lets discuss it further after this PR has been finalised 😄